tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaType1.scala

```diff
@@ -98,19 +98,69 @@ case object StrT1 extends TlaType1 {
 }
 
 /**
- * An uninterpreted type constant.
+ * <p>An uninterpreted type constant such as PROC_NAME.</p>
+ *
+ * <p>Inside the type checker, this class may also represent references to type aliases. Do not use this feature outside
+ * of the type checker.</p>
  *
  * @param name
  *   unique name of the constant type
  */
 case class ConstT1(name: String) extends TlaType1 {
-  require(name.forall(c => c.isUpper || c.isDigit || c == '_'),
-      "ConstT1 accepts identifiers in upper case, found: " + name)
+  require(ConstT1.isUninterpreted(name) || ConstT1.isAliasReference(name),
+      "ConstT1 accepts identifiers in upper case or $aliasReference, found: " + name)
 
   override def toString: String = name
 
   override def usedNames: Set[Int] = Set.empty
+}
+
+/**
+ * A companion object for [[ConstT1]].
+ */
+object ConstT1 {
```

Comment:
Not something I necessarily want now, but consider combining this class with `ModelValueHandler` (more accurately, moving methods from that class to here).
We don't want 2 unrelated classes handling utility for uninterpreted literals.

Reply:
Fixing the regular expression. I will open an issue about ModelValueHandler.